docs/guide/installation-guide.txt

```diff
@@ -1,3 +1,128 @@
 ==================
 Installation Guide
 ==================
+
+The recommended way to get started using the Node.js driver is by using ``NPM`` (Node Package Manager) to install the dependency in your project.
+
+MongoDB Driver
+--------------
+
+After you've created your project with ``npm init`` , you can install the MongoDB driver and its dependencies with the command:
+
+.. code-block:: sh
+
+   npm install mongodb --save
+
+This will download the MongoDB driver and add a dependency entry in your ``package.json`` file.
+
+Troubleshooting
+---------------
+
+The MongoDB driver depends on several other packages, including:
+
+* `bson <https://www.npmjs.com/package/bson>`_
+* `require_optional <https://www.npmjs.com/package/require_optional>`_
+* `safe-buffer <https://www.npmjs.com/package/safe-buffer>`_
+* `saslprep <https://www.npmjs.com/package/saslprep>`_
+
+Additionally, there are multiple optional dependencies that can be installed alongside the driver:
+
+* `bson-ext <https://www.npmjs.com/package/bson-ext>`_
+* `kerberos <https://www.npmjs.com/package/kerberos>`_
+* `mongodb-client-encryption <https://www.npmjs.com/package/mongodb-client-encryption>`_
+* `snappy <https://www.npmjs.com/package/snappy>`_
+
+These optional modules are all native C++ extensions. They are optional extensions and are not required for the driver to function.
+
+bson-ext Module
+---------------
+
+The ``bson-ext`` module is an alternative **BSON** parser that is written in C++. If you wish to use the ``bson-ext`` module you will need to add the ``bson-ext`` module to your module's dependencies.
+
+.. code-block:: sh
+
+   npm install bson-ext --save
+
+kerberos Module
+---------------
+
+If you need to use ``kerberos`` module you will need to add the ``kerberos`` module to your module's dependencies.
+
+.. code-block:: sh
+
+   npm install kerberos --save
+
+The ``kerberos`` package is a C++ extension that requires a build environment to be installed on your system. You must be able to build Node.js itself to be able to compile and install the ``kerberos`` module. Furthermore the ``kerberos`` module requires the MIT Kerberos package to correctly compile on UNIX operating systems. Consult your UNIX operating system package manager for what libraries to install.
+
+.. note::
+
+   Windows already contains the SSPI API used for Kerberos authentication. However, you will need to install a full compiler toolchain
+   using Visual Studio to correctly install the ``kerberos`` extension.
+
+Diagnosing on UNIX
```

Comment:
please remove these whole sections about diagnosis, they are covered in the kerberos README